src/App.tsx

```diff
@@ -31,65 +31,104 @@ export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+  let visibleGoods = [...goods];
+
+  if (sortType === 1 && isReversed === false) {
+    visibleGoods = visibleGoods.sort((a, b) => {
+      return a.localeCompare(b);
+    });
+  } else if (sortType === 2 && isReversed === false) {
+    visibleGoods = visibleGoods.sort((a, b) => {
+      return a.length - b.length;
+    });
+  } else if (sortType === 1 && isReversed === true) {
+    visibleGoods = visibleGoods.sort((a, b) => {
+      return b.localeCompare(a);
+    });
+  } else if (sortType === 2 && isReversed === true) {
+    visibleGoods = visibleGoods.sort((a, b) => {
+      return a.length - b.length;
+    }).reverse();
+  } else if (sortType === 0 && isReversed === true) {
+    visibleGoods = visibleGoods.reverse();
+  }
 
-  // Sort and reverse goods if needed
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component {
+  state: State = {
+    isReversed: false,
+    sortType: 0,
+  }
+
+  render() {
+    const goods = getReorderedGoods(goodsFromServer, this.state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={this.state.sortType === 1
+              ? 'button is-info'
+              : 'button is-info is-light'}
```

Comment:
You can use the && operator however, this solution is equally good
I think that using the values from the SortType enum will be a better solution than 0, 1 ,2...
```className={`button is-info ${(sortType !== SortType.ALPHABET) && ('is-light')}`}```

Reply:
I think my solution is clearer than &&